### ISSUE TITLE
Request - transaction attributes and trans public/private flag

### ISSUE THREAD

**Issue by dustymc** [ArctosDB/arctos#8240]
Help us understand your request (check below):

- [x] other

**Describe what you're trying to do** 

Ref: https://github.com/ArctosDB/dev/issues/40

The vague plan has been to craft transaction attributes similar to https://arctos.database.museum/info/ctDocumentation.cfm?table=ctagent_attribute_type, so that individual attributes types can be declared public or private. This seems to line up well with @mkoo 's needs (possibly on behalf of some third party??), and a forthcoming issue which will request more transaction (just accession?) data on the ``/guid/`` pages. BUT: we already have a public/private flag for transactions, this would set up a place for conflicts to occur and just make the system overly complex. 

One possibility:

1. Declare that some "core" accession data (perhaps only the number - could that be problematic?) is always public
2. Create public attributes that anyone who wants to share more can use
3. Create private attributes for anything that anyone doesn't want to share
4. Migrate (lots of details to work out, but not here)
5. Drop ``trans.is_public_fg``

And better ideas? Quick resolution would be most helpful so that the linked dev issue can proceed. HELP!

**Comment by campmlc** [comment]
My only concern with providing the accession number is that some accessions may need to be encumbered, delayed or temporarily pulled to comply with permit requirements, terms of agreement for deposition, or legal issues. Making a number public implies the accessioning process has been completed. Not assigning a number prevents having a dashboard to accumulate documentation.

**Comment by dustymc** [comment]
This could only possibly ever matter after you've created the accession and cataloged at least one record under it. MSB collections are in my 'generate an accession number' scripts, I cannot understand how "2024.048.Mamm" (which doesn't exist yet, but will the next time someone clicks the create button...) has anything to do with any of that?? (If it really does, then perhaps https://www.uuidgenerator.net/version4 would provide better identifiers??) 

Please clarify.

**Comment by campmlc** [comment]
I threw that comment out there just to see if others may have similar concerns. In our case, the accession identifiers are not the problem. We have no trouble issuing the identifiers. If we have records that are pending resolution of any accession data, such as clarification of permit status, then they should be encumbered until that is resolved. Assuming encumbrances work as expected, this should not prevent this issue from moving forward.
